IdentityCore/src/cache/token/MSIDCredentialCacheItem.m

```diff
@@ -250,12 +255,14 @@ - (BOOL)matchesWithHomeAccountId:(nullable NSString *)homeAccountId
 - (BOOL)matchByEnvironment:(nullable NSString *)environment
         environmentAliases:(nullable NSArray<NSString *> *)environmentAliases
 {
-    if (environment && ![self.environment isEqualToString:environment])
+    if (environment && 
+        ![self.environment.msidNormalizedString isEqualToString:environment.msidNormalizedString])
     {
         return NO;
     }
 
-    if ([environmentAliases count] && ![self.environment msidIsEquivalentWithAnyAlias:environmentAliases])
+    if ([environmentAliases count] && 
+        ![self.environment.msidNormalizedString msidIsEquivalentWithAnyAlias:environmentAliases])
```

Comment:
I think msidIsEquivalentWithAnyAlias already normalizes environment.

Reply:
It does caseInsensitiveCompare, but it doesn't trim the whitespace, so I'm going to keep it